### ISSUE TITLE
tag not applied at the end of "build"

### ISSUE THREAD

**Issue by peterthomassen** [docker/compose#6807]
##  Description of the issue

Sometimes, when I run `docker-compose build`, the tag is not applied to the newly built image, but to the base image (`FROM` line in the Dockerfile).

This happens sporadically, every few weeks. I has happened once that I deployed the nginx base image because of that (instead of the custom image that I built and intended to deploy).

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.24.0, build 0aa59064
docker-py version: 3.7.2
CPython version: 3.6.8
OpenSSL version: OpenSSL 1.1.0j  20 Nov 2018
```

**Output of `docker version`**
```
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.1
 Git commit:        2d0083d
 Built:             Wed Jul  3 12:13:59 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.1
  Git commit:       2d0083d
  Built:            Mon Jul  1 19:31:12 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

When running `docker-compose build` on https://github.com/desec-io/desec-stack/blob/747fe7a95915a83fc8560230cc994455d05f56a0/docker-compose.yml, with https://github.com/desec-io/desec-static/blob/a2d436813b8cc34dce203b466ecbd80ccb034ec5/Dockerfile , the output is as follows:

## Steps to reproduce the issue

1. Create a directory and copy https://github.com/desec-io/desec-stack/blob/747fe7a95915a83fc8560230cc994455d05f56a0/docker-compose.yml (plus build directories of the services)
2. Create `static/` subdirectory and copy https://github.com/desec-io/desec-static/blob/a2d436813b8cc34dce203b466ecbd80ccb034ec5/Dockerfile there (plus other stuff in that directory)
3. `docker-compose build`, then `docker images`

### Observed result
The build proceeds as fllows:
```
Building static
[...]
Step 15/15 : RUN cp -a dist/. /usr/share/nginx/html && rm -rf /usr/src/app
 ---> Using cache
 ---> e1304267fde2
Successfully built e1304267fde2
Successfully tagged desec/dedyn-static:latest
```

Sporadically, however, the wrong image gets tagged:
```
$ docker images | grep static
desec/dedyn-static                      <none>              e1304267fde2        6 days ago          713MB
desec/dedyn-static                      latest              ac44715da54a        5 weeks ago         109MB
```

The image that got tagged is actually the nginx parent image:
```
$ docker images | grep nginx
nginx                                   stable              ac44715da54a        5 weeks ago         109MB
```

### Expected result
I expect that the following lines actually mean that `e1304267fde2` was tagged as `desec/dedyn-static:latest`.
```
Successfully built e1304267fde2
Successfully tagged desec/dedyn-static:latest
```

**Comment by ulyssessouza** [comment]
Hello @peterthomassen . I couldn't reproduce the bug :disappointed: 
Could you provide a smaller and deterministic version of your `Steps to reproduce the issue` ?

**Comment by peterthomassen** [comment]
@ulyssessouza As the bug happens only every few weeks, I have only had two chances at debugging so far. Until this moment, I only found that rerunning `docker-compose build` brings things bad to order. I have, currently, no idea whatsoever which circumstances would trigger this behavior, and I am not aware of a deterministic way to reproduce this (short of trying a lot of times).

Still, I consider this bug rather severe, so instead of closing it, I would advise reviewing the code between building an image and tagging it, which gets run as part of `docker-compose build`. For example, -- when is the image identified that will be tagged? Does this variable have another value before it is set to the to-be-tagged image? Could the previous value be the base image from the Dockerfile? (That would be consistent with what I observed.) Maybe it would help to introduce a separate variable, `image_to_be_tagged`, which is not written to, except after the last build step? ...
